src/Microsoft.AspNet.Mvc.Core/Formatters/MediaTypeComponent.cs

```diff
@@ -0,0 +1,138 @@
+        public bool HasName(string name)
+        {
+            return HasName(new StringSegment(name));
+        }
+
+        /// <summary>
+        /// Determines if the current <see cref="MediaTypeComponent"/> has the given name. The comparison is
+        /// case insensitive.
+        /// </summary>
+        /// <param name="name">The <paramref name="name"/> to compare to.</param>
+        /// <returns><code>true</code>if the name of this <see cref="MediaTypeComponent"/> is equal to the given
+        /// <paramref name="name"/>; otherwise <code>false</code>.</returns>
+        public bool HasName(StringSegment name)
+        {
+            return Name.Equals(name, StringComparison.OrdinalIgnoreCase);
+        }
+
+        /// <summary>
+        /// Determines if the current <see cref="MediaTypeComponent"/> has the given value. The comparison is
+        /// case insensitive.
+        /// </summary>
+        /// <param name="value">The <paramref name="value"/> to compare to.</param>
+        /// <returns><code>true</code>if the value of this <see cref="MediaTypeComponent"/> is equal to the given
+        /// <paramref name="value"/>; otherwise <code>false</code>.</returns>
+        public bool HasValue(string value)
+        {
+            return HasValue(new StringSegment(value));
+        }
+
+        /// <summary>
+        /// Determines if the current <see cref="MediaTypeComponent"/> has the given value. The comparison is
+        /// case insensitive.
+        /// </summary>
+        /// <param name="value">The <paramref name="value"/> to compare to.</param>
+        /// <returns><code>true</code>if the value of this <see cref="MediaTypeComponent"/> is equal to the given
+        /// <paramref name="value"/>; otherwise <code>false</code>.</returns>
+        public bool HasValue(StringSegment value)
+        {
+            return Value.Equals(value, StringComparison.OrdinalIgnoreCase);
+        }
+
+        /// <summary>
+        /// Indicates whether the current object is equal to another object of the same type.
+        /// </summary>
+        /// <param name="other">An object to compare with this object.</param>
+        /// <returns><code>true</code> if the current object is equal to the other parameter; otherwise, <code>false</code>.</returns>
+        public bool Equals(MediaTypeComponent other)
+        {
+            return HasName(other.Name) && HasValue(other.Value);
+        }
+
+        /// <inheritdoc />
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(null, obj))
+            {
+                return false;
+            }
+
+            return obj is MediaTypeComponent && Equals((MediaTypeComponent)obj);
+        }
+
+        /// <inheritdoc />
+        public override int GetHashCode()
+        {
+            HashCodeCombiner hashCode = HashCodeCombiner.Start();
+            hashCode.Add(Name.Value);
+            hashCode.Add(Value.Value);
```

Comment:
Yes they do, but we don't use MediaTypeComponent in a way that involves getting it's hash code, and if in the future we do, we can easily change the implementation.

For what is worth, I'm happy just doing return 1 which satisfies the GetHashCode contract and  causes a degenerate list in the worst case scenario. The only reason GetHashCode is overriden is because it needs to be done for equality